topo/node/cptx/cptx_test.go

```diff
@@ -44,6 +48,33 @@ func (f *fakeWatch) ResultChan() <-chan watch.Event {
 	return eCh
 }
 
+// removeCommentsFromConfig removes comment lines from a JunOS config file
+// and returns the remaining config in an io.Reader.
+// Using scrapl_cfg_testing results in an EOF error when config includes comments.
+// Comments in config files are not problematic when using kne_cli (not testing).
+// This is a simple implementation that only removes lines that are entirely comments.
+func removeCommentsFromConfig(f *os.File) (io.Reader, error) {
+	var buf bytes.Buffer
+	br := bufio.NewReader(f)
+	regexp := regexp.MustCompile(`^\s*(?:(?:\/\*)|[#\*])`)
+	for {
+		line, err := br.ReadBytes('\n')
+		if err != nil && err != io.EOF {
+			return nil, fmt.Errorf("br.ReadBytes() failed: %+v\n", err)
+		}
+
+		if regexp.Find(line) != nil {
+			continue
+		}
+		fmt.Fprint(&buf, string(line))
+
+		if err == io.EOF {
```

Comment:
would it make more sense to check this right after (or even in the initial err check)?

```
line, err := ...
if err != nil {
  if err == io.EOF {
    break
  }
  return nil, err
}
```

then

```
if re.Find(line) == nil {
  fmt.Fprint(&buf, string(line))
}
```

Those changes should reduce the number of control flow statements and make it slightly more obvious of what the code is doing. What are your thoughts?

Reply:
Looking at this once more, I realized I can still use the change below regardless of how we handle the error:

```
if re.Find(line) == nil {
  fmt.Fprint(&buf, string(line))
}
```

Reply:
Oh I wasn't aware of that behavior, good with me then